include/myst/thread.h

```diff
@@ -292,8 +314,13 @@ MYST_INLINE bool myst_valid_thread(const myst_thread_t* thread)
 }
 
 myst_thread_t* myst_thread_self(void);
+myst_process_t* myst_process_self(void);
 
-void myst_zombify_thread(myst_thread_t* thread);
+void myst_zombify_process(myst_process_t* process);
+MYST_INLINE bool myst_is_zombied_process(myst_process_t* process)
+{
+    return (process->main_process_thread == NULL) ? true : false;
```

Comment:
For my understanding, the `myst_process_t` associated with the zombied process is free'd by the parent?

Reply:
process is only freed in myst_wait and during zombie shutdown

Reply:
So what you say can be interpreted as only the parent, as only the parent can do a waitpid() on a child process.